README.md

```diff
@@ -72,6 +74,35 @@ Source this environment in and provide the password when prompted:
 $ source .env
 ```
 
+## Jupyter notebooks
+Though Jupyter notebooks and lab are well suited to running Visier connector code, the use of OS-level environment variables may not be ideal for many users in this setting. For this purpose and since version `0.9.9`, the connector supports [dotenv](https://pypi.org/project/python-dotenv/) to facilitate a more dynamic switching of Visier authentication parameters.
+
+### Jupyter OAuth2 example
+In order to authenticate using OAuth, you will need a registered client application configuration in your tenant. Administrator-level privileges are required in order to register OAuth clients. 
+
+Create an environment file to store the authentication parameters. If the file is called `.env`, the Python package `dotenv` will attempt to load that. If given an arbitrary name, that named has to be provided when loading the environment using `dotenv`. 
+
+Example environment file:
+```
+VISIER_HOST=https://customer-specific.api.visier.io
+VISIER_CLIENT_ID=client-id-from-registration
+VISIER_APIKEY=the-api-key-issued-by-visier
+```
+
+Create an OAuth authentication object as per the following snippet:
+```python
+from dotenv import dotenv_values
+from visier.connector import VisierSession, make_auth
+from visier.api import QueryApiClient
+
+env_creds = dotenv_values()
+auth = make_auth(env_values=env_creds)
+
+with VisierSession(auth) as s:
+    query_client = QueryApiClient(s)
+    ...
+```
+
 ## Connector Separation
 As of version `0.9.5`, the Python connector has separated the API calls from the `VisierSession` object. As a result of this change, the query execution methods on the `VisierSession` have been deprecated and will be subject to removal in a future release.
```

Comment:
As of version `0.9.5`, the Python connector separates API calls from the `VisierSession` object. As a result of this change, query execution methods on the `VisierSession` are deprecated and will be removed in a future release